### DISCUSSION TITLE
Error reporting in scoped tasks

### DISCUSSION THREAD

**Discussion by jplatte** [rayon-rs/rayon#1104] [General]
Hello, and thanks a lot for Rayon!

~~This issue is more like a question, but GitHub discussions aren't enabled here and I didn't want to use a synchronous communication medium.~~

I'm currently using scoped tasks (`Scope::spawn`) for the first time and am wondering if there are any established patterns for reporting errors that happen in those. I think I want to collect all the errors and print them after the scope ends. I guess is easy enough with an MPSC channel? That's what I will try first in any case. I'm just wondering: Are there any established error-handling patterns for scoped tasks? Is this discussed in the documentation somewhere and I just missed it?

**Comment by cuviper** [comment]
> GitHub discussions aren't enabled here

OK, I've flipped that switch so we can try it out!

To the actual question -- I'm not aware of any particular patterns for this, but I think a channel should be fine, depending on how you handle the receiving end. For example, if you block on `recv()` directly within the `scope` closure, then that rayon thread will be stuck on this task alone. Maybe you could `try_recv()` and then `yield()`, but that's a busy loop chewing time if there's nothing else to be done. Or you could just `recv` everything after the scope has ended.

**Reply by jplatte** [reply]
> Or you could just recv everything after the scope has ended.

Yeah, that's what I implemented, use a channel with no limit on buffer size and only look at it after all tasks have ended. That means no short circuiting but it's fine for my use case.
